src/fides/api/ctl/routes/generate.py

```diff
@@ -192,11 +200,15 @@ def generate_aws(
             status_code=status.HTTP_401_UNAUTHORIZED,
             detail=str(error),
         )
+    if generate_type == "systems":
+        log.info("Generating systems from AWS")
+        aws_resources = generate_aws_systems(
+            organization=organization, aws_config=aws_config
+        )
+    else:
+        aws_resources = [generate_dynamo_db_datasets(aws_config=aws_config)]
```

Comment:
It's a bit confusing for the `generate_aws` function to return lists of `Datasets` and `Systems`. I see the type hint on the function allows for this but should we tighten up the types here and move the dataset generation to a new function that returns datasets?

Reply:
Totally agree 👍🏽 changed this up as part of 5bc94373add7218f9f2aeac5afd2ceb7ab85351d as well